src/presentation/user.dto.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * @fileoverview In the context of NestJS, DTO is a class represents the value given to Controller.
+ * Each property need to be decorated with class-validator decorators, to validate user input.
+ *
+ * @see {@link https://docs.nestjs.com/techniques/validation}
+ */
+
+import { UserId } from 'src/domain/user.interface';
+import {
+  IsDateString,
+  IsNotEmpty,
+  IsNumberString,
+  IsOptional,
+  IsString,
+  Length,
+} from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+
+export class UserDto {
```

Comment:
It seems the `UserDto` has the exact data shape with [NewUser](https://github.com/Wiredcraft/test-backend/pull/63/files#diff-a5105e8e39b26a4054abe7d84a51b24b045afb9789a26d03623f84313fdbc35cR12) except the decoration with class-validator.
Any idea to avoid duplication?

Reply:
It's intentionally separated.

DTO and Entity have different responsibilities and belong to different tiers. If we mix them, controllers can use business logic in an entity like `follow(User)`, and domain models depend on things required from the presentation tier like validation. By separating them, we can make the responsibility of classes clear, and keep the code base safe to change.